pdns/python_modules/pdns.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import sys
+import time
+import copy
+import socket
+
+descriptors = list()
+Desc_Skel   = {}
+Debug = False
+
+METRICS = {
+    'time' : 0,
+    'data' : {}
+}
+
+LAST_METRICS = copy.deepcopy(METRICS)
+METRICS_CACHE_MAX = 5
+
+def dprint(f, *v):
+    if Debug:
+        print >>sys.stderr, "DEBUG: "+f % v
+
+def get_metrics():
+    '''Return all metrics'''
+
+    global METRICS, LAST_METRICS, params
+
+    if (time.time() - METRICS['time']) > METRICS_CACHE_MAX:
+
+        new_metrics = {}
+
+        s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        s.connect(params['pdns_socket'])
+        s.send('SHOW *\n')
+        output = s.recv(1024)
+
+        for i in (filter(None, output.split(','))):
+            pair = i.split('=')
+            new_metrics[pair[0]] = pair[1]
+
+        LAST_METRICS = copy.deepcopy(METRICS)
+        METRICS = {
+            'time': time.time(),
+            'data': new_metrics
+        }
+
+    return [METRICS, LAST_METRICS]
+
+def get_delta(name):
+    '''Return change over time for the requested metric'''
+
+    # get metrics
+    [curr_metrics, last_metrics] = get_metrics()
+    name = name.replace('pdns_', '')
+
+    try:
+        delta = ((int(curr_metrics['data'][name]) - int(last_metrics['data'][name]))/(curr_metrics['time'] - last_metrics['time']))
+        if delta < 0:
+            delta = 0
+    except KeyError:
+        delta = 0.0
+
+    return delta
+
+
+def metric_init(param):
+    global descriptors, Desc_Skel, Debug, params
+    params = copy.deepcopy(param)
+
+    descriptors = []
+
+    print '[pdns] pdns'
```

Comment:
Could you make sure that this isn't dumping stuff to console during metric_init unless debug is enabled?